pandas/compat/numpy/__init__.py

```diff
@@ -68,6 +68,22 @@ def np_array_datetime64_compat(arr, *args, **kwargs):
     return np.array(arr, *args, **kwargs)
 
 
+# np.percentile should support datetime64 in upsteam. numpy/numpy#11620
+# TODO: change the version if the bug fixed
+if _nlv >= '99.99.99':
+    np_percentile_compat = np.percentile
+else:
+    def np_percentile_compat(values, q, **kw):
```

Comment:
the iNat concern aside, this looks like a nice function.  But I don't see why it should be in compat.  Might make more sense in nanops?